### PR TITLE
PhysicsBehavior bits

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/game/logic/object/object.cpp [hunk 1]
```diff
@@ -202,3 +203,87 @@ void Object::Clear_Model_Condition_State(ModelConditionFlagType a)
...
+}
+
+void Object::Defect(Team *team, unsigned int i)
+{
+#ifdef GAME_DLL
+    Call_Method<void, Object, Team *, unsigned int>(PICK_ADDRESS(0x0054EEC0, 0x007D924B), this, team, i);
+#endif
+}
+
+char Object::Get_Crushable_Level() const
```
Comment (tomsons26) [inline src/game/logic/object/object.cpp]: crushable and crusher return unsigned char according to rts map symbols, the member in template should be checked as well

Comment (jonwil) [reply src/game/logic/object/object.cpp]: Fixed. The functions in ThingTemplate are already returning unsigned char so it's all good there.

File: src/game/logic/object/object.cpp [hunk 2]
```diff
@@ -202,3 +203,87 @@ void Object::Clear_Model_Condition_State(ModelConditionFlagType a)
...
+#endif
+}
+
+void Object::On_Collide(Object *other, const Coord3D *loc, const Coord3D *normal)
+{
+    for (BehaviorModule **module = m_allModules; *module != nullptr; module++) {
+        CollideModuleInterface *collide = (*module)->Get_Collide();
+
+        if (collide != nullptr) {
+            if (Get_Status_Bits().Test(OBJECT_STATUS_NO_COLLISIONS)) {
```
Comment (tomsons26) [inline src/game/logic/object/object.cpp:260]: was break i guess rather?

Comment (jonwil) [reply src/game/logic/object/object.cpp:260]: IDA pseudo showed return. And return seems to fit.

File: src/game/logic/system/damage.h [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    virtual void CRC_Snapshot(Xfer *xfer) override {}
+    virtual void Xfer_Snapshot(Xfer *xfer) override;
+    virtual void Load_Post_Process() override {}
+
+    ObjectID m_sourceID;
+    ThingTemplate *m_thingTemplate;
+    uint16_t m_playerMask;
+    DamageType m_damageType;
+    ObjectStatusTypes m_damageStatusType;
+    unsigned int m_unk1; // seems to be some kind of enum
```
Comment (tomsons26) [inline src/game/logic/system/damage.h]: comment that it defaults to 11, and is set to 9 in PoisonedBehavior::Update

File: src/game/logic/system/damage.h [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    DeathType m_deathType;
+    float m_amount;
+    bool m_unk2;
```
Comment (tomsons26) [inline src/game/logic/system/damage.h]: comment that its only set to true in Object::Kill

Comment (jonwil) [reply src/game/logic/system/damage.h]: fixed.

File: src/game/logic/system/damage.h (hunk 3 above)
Comment (jonwil) [reply src/game/logic/system/damage.h]: fixed.

**Review by tomsons26 — APPROVED** [review]

File: src/game/logic/object/update/aiupdate.cpp [hunk 5]
```diff
@@ -44,3 +44,8 @@ float AIUpdateInterface::Get_Cur_Locomotor_Speed() const
     return 0.0f;
 #endif
 }
+
+bool AIUpdateInterface::Has_Locomotor_For_Surface(LocomotorSurfaceType t)
+{
+    return m_locomotorSet.Find_Locomotor(t) != nullptr;
+}
```
Comment (xezon) [inline src/game/logic/object/update/aiupdate.cpp]: Can add EOL

File: src/game/logic/system/damage.cpp [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "damage.h"
+#include "thingfactory.h"
+
+void DamageInfoInput::Xfer_Snapshot(Xfer *xfer)
+{
+    unsigned char current_version = 3;
+    unsigned char version = 3;
```
Comment (xezon) [inline src/game/logic/system/damage.cpp:21]: uint8_t

File: src/game/logic/system/damage.cpp [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+        if (xfer->Get_Mode() == XFER_LOAD) {
+            m_thingTemplate = g_theThingFactory->Find_Template(name, true);
+        }
+    }
+}
+
+void DamageInfoOutput::Xfer_Snapshot(Xfer *xfer)
+{
+    unsigned char version = 1;
```
Comment (xezon) [inline src/game/logic/system/damage.cpp:61]: uint8_t

File: src/game/logic/system/damage.cpp [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    xfer->xferVersion(&version, 1);
+    xfer->xferReal(&m_actualDamageDealt);
+    xfer->xferReal(&m_actualDamageClipped);
+    xfer->xferBool(&m_noEffect);
+}
+
+void DamageInfo::Xfer_Snapshot(Xfer *xfer)
+{
+    unsigned char version = 1;
```
Comment (xezon) [inline src/game/logic/system/damage.cpp:70]: uint8_t

File: src/game/logic/object/update/physicsupdate.h [hunk 9]
```diff
@@ -28,6 +27,36 @@ enum PhysicsTurningType
     TURN_POSITIVE,
 };
 
+class PhysicsBehaviorModuleData : public UpdateModuleData
+{
+public:
+    PhysicsBehaviorModuleData() {}
```
Comment (xezon) [inline src/game/logic/object/update/physicsupdate.h]: None of the members are initialized?

File: src/game/logic/object/update/aiupdate.cpp (hunk 5 above)
Comment (jonwil) [reply src/game/logic/object/update/aiupdate.cpp]: fixed.

File: src/game/logic/object/update/physicsupdate.h (hunk 9 above)
Comment (jonwil) [reply src/game/logic/object/update/physicsupdate.h]: fixed.

**Pull request merged**